### ISSUE TITLE
딥러닝 워크북

### ISSUE THREAD

**Issue by wooaust** [codingeverybody/codingyahac#992]
우선 강의 잘 보고 있습니다. 

워크북 관련 질문이 있어 글을 올립니다.

제가 이해 하기론, dt는 weight의 작은 변화에 loss가 어떻게 움직이는지 보기 위해 작은 값(0.0001)을 임의적으로 선택한 것으로 보이는데 next W을 구할 때 0.001이 사용된 이유가 궁금합니다. 이 것도 임의로 선택된 값인지 알고 싶습니다. 또, 더 큰값(0.001) 이 사용된 이유가 weight을 좀 더 많이 변화시키기 이유인지도 궁금합니다. 

이 값들이 tensorflow에서 사용되는 default값 인지도 궁금합니다.

감사합니다.

**Comment by jiyongjung0** [comment]
안녕하세요, 말씀하신대로 dt가 weight나 bias를 살짝 바꿔보는 양으로 선택된 것 같습니다. 그런데 강의 내용으로는 모두 0.0001인 것 같은데 0.001은 어느 부분을 언급하신 것인지 좀 더 자세히 알려주실 수 있을까요? (강의 번호 및 동영상의 몇 분 몇 초쯤이라든가...)

그리고 실제 딥러닝 모델을 만들 때는 가능하면 수학적으로 미분 가능한 형태로 만들어서 미분한 수식(즉 loss가 어떻게 움직이는지 바로 계산할 수 있는 수식)에 값을 대입하여 사용합니다. 물론 엄밀하게 따지면 더 복잡한 내용이 있고 저도 잘 모릅니다만.. ^^;;; 텐서플로나 파이토치 등은 자동으로 미분을 해주는 기능을 갖추고 있기 때문에 잘 몰라도 어느 정도는 머신 러닝 모델을 만들 수 있답니다.

**Comment by jiyongjung0** [comment]
다시 워크북을 보니 수식에 말씀하신 대로 0.001 이 들어있네요. 이 값은 보통 러닝레이트(learning rate) 라고 부르는 값으로 보아야할 것 같은데요. 쉽게 말해서 loss가 움직이는 경향을 보고 다음에 어느 쪽으로 가야할지 알아냈는데, 그렇다고 이렇게 학습한 내용이 완전히 정확한 값일리는 없기 때문에 그걸 받아들이는 정도를 조정을 하게 됩니다. 러닝레이트는 머신 러닝 모델을 만들 때마다 모두 제각각이기 때문에 기본 값이라고 할 것이 없습니다. 말씀하신대로 더 큰값을 사용하면 weight가 더 많이 변화해서 답을 향해 더 빠르게 간다는 장점이 있지만, 너무 확확 움직여서 답에 아주 가까이 가지는 못하게 되거나 아예 헤매게 되는 경우도 있습니다. 그래서 모델을 만들 때마다 좋은 러닝레이트를 찾아서 이 값 저 값을 넣어보는 과정을 거치게 됩니다.

**Comment by blackdew** [comment]
네. 이번 수업에서 learning rate를 수업범위에 넣지 않았습니다. 정말 마지막까지 고민한 부분이었네요. :) 
실제 모델이 학습하는 방법을 정의를 하게 되어있는데요. 해당 알고리즘 방법들을 optimizer라고 해요. 

워크북은 optimizer 중에 알고리즘 중에 가장 기본이 되는 
'경사하강법(Gradient Decent Algorithm)'을 기반으로 작성이 되어있는데요.

tensorflow에서는 default 알고리즘이 Adam으로 되어있고, earning rate의 기본값도 0.001로 되어있습니다. 
model.compile 하는 부분에서 optimizer를 조정해 줄 수 있습니다.

**Comment by wooaust** [comment]
감사합니다.
